dali/pipeline/operators/decoder/nvjpeg_decoder_new.h

```diff
@@ -0,0 +1,268 @@
+    ******************************************/
+
+    for (int i = 0; i < batch_size_; i++) {
+      const auto &in = ws->Input<CPUBackend>(0, i);
+      const auto *input_data = in.data<uint8_t>();
+      const auto in_size = in.size();
+      // Get necessary image information
+
+      nvjpegStatus_t ret = nvjpegJpegStreamParse(handle_,
+                                                 static_cast<const unsigned char*>(input_data),
+                                                 in_size,
+                                                 1,
+                                                 1,
+                                                 jpeg_streams_[i]);
+
+      if (ret == NVJPEG_STATUS_BAD_JPEG) {
+        auto file_name = in.GetSourceInfo();
+        try {
+          EncodedImageInfo info;
+          const auto image = ImageFactory::CreateImage(
+                                            static_cast<const uint8 *>(input_data), in_size);
+          const auto dims = image->GetImageDims();
+          info.heights[0] = std::get<0>(dims);
+          info.widths[0] = std::get<1>(dims);
+          info.nvjpeg_support = false;
+          output_info_[i] = info;
+        } catch (const std::runtime_error &e) {
+          DALI_FAIL(e.what() + "File: " + file_name);
+        }
+      } else {
+        EncodedImageInfo info;
+        NVJPEG_CALL(nvjpegJpegStreamGetFrameDimensions(jpeg_streams_[i],
+                                                       info.widths,
+                                                       info.heights));
+        NVJPEG_CALL(nvjpegJpegStreamGetComponentsNum(jpeg_streams_[i],
+                                                           &info.c));
+        NVJPEG_CALL(nvjpegJpegStreamGetChromaSubsampling(jpeg_streams_[i],
+                                                         &info.subsampling));
+        info.nvjpeg_support = SupportedSubsampling(info.subsampling);
+        output_info_[i] = info;
+
+        if (ShouldBeHybrid(info)) {
+          image_decoders_[i] = decoder_huff_hybrid_;
+        } else {
+          image_decoders_[i] = decoder_huff_host_;
+        }
+
+        /*
+        // TODO(spanev): add this function when integrating Crop
+        nvjpegnvjpegDecodeParamsSetROI(decode_params_[pos], offset_x, offset_y, roi_w, roi_h);
+        */
+        NVJPEG_CALL(nvjpegStateAttachPinnedBuffer(image_states_[i],
+                                                  pinned_buffer_[i]));
+        NVJPEG_CALL(nvjpegDecodeJpegHost(
+            handle_,
+            image_decoders_[i],
+            image_states_[i],
+            decode_params_[i],
+            jpeg_streams_[i]));
+      }
+    }
+
+    /******************************************
+     ****************** GPU *******************
+    ******************************************/
+
+    std::vector<Dims> output_shape(batch_size_);
+    // Creating output shape and setting the order of images so the largest are processed first
```

Comment:
When it is separated and the worker are executed by the CPU executor worker thread pool then this sorting won't make any difference.

Reply:
Also user need to remember to set affinity to true for the executor as we will no longer have affinity in the nvJPEG istels.

Reply:
In the splitted part, this operator will be executed by the Mixed executor

Reply:
I don't know if we want to leave it as mixed - @klecki how it will work with prefetch queue separation?

Reply:
AFAIK and from what we discussed, the only operators accepting CPU input and producing GPU output will remain the Mixed operators

Reply:
Hence the current design of CPUBackend for CPU phase and MixedBackend for GPU phase (just pushed the two-stage operator)

Reply:
Still MixedBackend for GPU will be for GPU part while we CPU part may require affinity as well. Also sorting images by size will be gone - we need to test how it will behave.

Reply:
Maybe we can add a note about the executor affinity in the Op docstring